include/xtensor/xfixed.hpp

```diff
@@ -16,6 +16,7 @@
 #include <utility>
 #include <vector>
 
+#include "xadapt.hpp"
```

Comment:
Which function defined in `xadapt.hpp` do we call in `xfixed.hpp`?

Reply:
Let me paste my error-stream here:
```
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:30:16: error: no template named 'sequence_builder'; did you mean
      '::xtl::detail::sequence_builder'?
        struct sequence_builder<xt::const_array<T, N>>
               ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xshape.hpp:72:16: note: '::xtl::detail::sequence_builder' declared here
        struct sequence_builder;
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:30:16: error: class template partial specialization of 'sequence_builder'
      not in a namespace enclosing 'detail'
        struct sequence_builder<xt::const_array<T, N>>
               ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xshape.hpp:72:16: note: explicitly specialized declaration is here
        struct sequence_builder;
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:51:34: error: unknown type name 'layout_type'; did you mean
      '::xt::layout_type'?
    template <class ET, class S, layout_type L, bool SH, class Tag>
                                 ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:94:19: error: unknown type name 'layout_type'; did you mean
      '::xt::layout_type'?
        template <layout_type L, std::size_t I, std::size_t... X>
                  ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:98:33: error: use of undeclared identifier 'layout_type'; did you mean
      '::xt::layout_type'?
        struct calculate_stride<layout_type::column_major, I, Y, X...>
                                ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:100:74: error: use of undeclared identifier 'layout_type'; did you mean
      '::xt::layout_type'?
            constexpr static std::ptrdiff_t value = Y * calculate_stride<layout_type::column_major, I - 1, X...>::value;
                                                                         ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:104:33: error: use of undeclared identifier 'layout_type'; did you mean
      '::xt::layout_type'?
        struct calculate_stride<layout_type::column_major, 0, Y, X...>
                                ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:112:53: error: no template named 'at'; did you mean '::xt::detail::at'?
            constexpr static std::ptrdiff_t value = at<sizeof...(X) - I, X...>::value * calculate_stride_row_major<I - 1, X...>::value;
                                                    ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xshape.hpp:290:16: note: '::xt::detail::at' declared here
        struct at
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:122:33: error: use of undeclared identifier 'layout_type'; did you mean
      '::xt::layout_type'?
        struct calculate_stride<layout_type::row_major, I, X...>
                                ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:129:23: error: unknown type name 'layout_type'; did you mean
      '::xt::layout_type'?
            template <layout_type L, size_t I, class SEQ>
                      ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:132:23: error: unknown type name 'layout_type'; did you mean
      '::xt::layout_type'?
            template <layout_type L, size_t I, size_t... X>
                      ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:138:23: error: unknown type name 'layout_type'; did you mean
      '::xt::layout_type'?
            template <layout_type L, size_t I, class SEQ >
                      ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:145:19: error: unknown type name 'layout_type'; did you mean
      '::xt::layout_type'?
        template <layout_type L, class R, std::size_t... X, std::size_t... I>
                  ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:146:44: error: no template named 'fixed_shape' in namespace
      'GMatTensor::Cartesian3d::xt'; did you mean '::xt::fixed_shape'?
        constexpr R get_strides_impl(const xt::fixed_shape<X...>& shape, std::index_sequence<I...>)
                                           ^~~~
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xshape.hpp:35:11: note: '::xt::fixed_shape' declared here
    class fixed_shape;
          ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:148:33: error: use of undeclared identifier 'layout_type'; did you mean
      '::xt::layout_type'?
            static_assert((L == layout_type::row_major) || (L == layout_type::column_major),
                                ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:148:66: error: use of undeclared identifier 'layout_type'; did you mean
      '::xt::layout_type'?
            static_assert((L == layout_type::row_major) || (L == layout_type::column_major),
                                                                 ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:191:35: error: no template named 'fixed_shape' in namespace
      'GMatTensor::Cartesian3d::xt'; did you mean '::xt::fixed_shape'?
        struct fixed_compute_size<xt::fixed_shape<X...>>
                                  ^~~~
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xshape.hpp:35:11: note: '::xt::fixed_shape' declared here
    class fixed_shape;
          ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:219:15: error: unknown type name 'layout_type'; did you mean
      '::xt::layout_type'?
    template <layout_type L, class R, std::size_t... X>
              ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xlayout.hpp:21:16: note: '::xt::layout_type' declared here
    enum class layout_type
               ^
In file included from /Users/tdegeus/data/prog/src/GMatTensor/test/Cartesian3d.cpp:4:
In file included from /Users/tdegeus/data/prog/src/GMatTensor/include/GMatTensor/Cartesian3d.h:26:
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xfixed.hpp:220:35: error: no template named 'fixed_shape'; did you mean
      '::xt::fixed_shape'?
    constexpr R get_strides(const fixed_shape<X...>& shape) noexcept
                                  ^
/Users/tdegeus/opt/miniconda3/envs/test/include/xtensor/xshape.hpp:35:11: note: '::xt::fixed_shape' declared here
    class fixed_shape;
          ^
```

Reply:
Oh, I think including `<xtl/xsequence.hpp>` should fix the issue.

Reply:
Is there a way to check in the CI, or do we believe it?

Reply:
The compiler is complaining about not finding the definition of `sequence_builder` which is defined in `xtl/xsequence.hpp`. This file is included by `xadapt.hpp`.

Also notice that `xadapt.hpp` already includes `xfixed.hpp`, therefore you cannot include `xadapt.hpp` in `xfixed.hpp`.

Reply:
Checking in the CI would requires splitting the file `test_xfixed.cpp` in many files and include only what we need in each of these new files.